File: Python Programming/problems on string manipulation/problem112.py
'''
112. Write a program which accept string from user and copy the contents of that string into another string. (Implement strncpy() function) 
	
	Input :   "Marvellous Multi OS"    10 
	Output :   " Marvellous "
	Note : If third parameter is greater than the size of source string then copy whole string into destination. 
	
	Author : Milind Nalawade
	
	Date : 16/07/2021
	
	Language : C programming

'''
#====
#Logic
#====
def StringCopy(str,no):
	return str[0:no]

#=============
#Entry point function
#=============
def main():
	str=input("\n Enter your string : ")
	no=int(input("\n Enter your number : "))
	dest=StringCopy(str,no)
	print("\n Output : {} \n".format(dest))
	
#=====
#starter
#=====
if __name__=="__main__":
	main()

File: C Programming/problems on strings maipulation/problem112.c
/*
112. Write a program which accept string from user and copy the contents of that string into another string. (Implement strncpy() function) 
	
	Input :   "Marvellous Multi OS"    10 
	Output :   " Marvellous "
	Note : If third parameter is greater than the size of source string then copy whole string into destination. 
	
	Author : Milind Nalawade
	
	Date : 16/07/2021
	
	Language : C programming
	
*/
#include<stdio.h>
//Logic
void CopyString(char* arr,char* brr)
{
	int no=0;
	printf("\n Enter your number : ");
	scanf("%d",&no);
	for(int i=0;i<no;i++)
	{
		*brr=*arr;
		*arr++;
		*brr++;
	}
	*brr='\0';
}

//Entry point function
int main()	
{
	char arr[30];
	char brr[30];
	printf("\n Enter your string : ");
	scanf("%[^'\n']s",&arr);
	CopyString(arr,brr);
	printf("\n Output : %s\n",brr);
	return 0;
}